src/index.js

```diff
@@ -1,38 +1,55 @@
-import { renderProjects, renderTasks } from "./dom";
-import { projectModal, submitProject, allProjects } from "./project";
+import { projectModal, submitProject } from "./project";
 import { submitTask, taskModal } from "./task";
-import { checkingTasks } from "./clear";
+import { getSelectedProject, selectProject, state } from "./project/state";
+import { renderProjects, renderSelectedProject } from "./project/render";
+import { renderTasks } from "./task/render";
+import { clearTasks } from "./task/state";
 
 let projectBtn = document.getElementById("projectBtn");
 let projectSubmitBtn = document.getElementById("projectSubmitBtn");
 let taskBtn = document.getElementById("taskBtn");
-let taskSubmitBtn = document.getElementById('taskSubmitBtn')
-let clearBtn = document.getElementById('clearBtn')
+let taskSubmitBtn = document.getElementById("taskSubmitBtn");
+let clearBtn = document.getElementById("clearBtn");
 
 // Makes project modal appear
 projectBtn.addEventListener("click", () => {
   projectModal();
 });
 
+const renderTaskWithHandler = (project) => {
+  // This can be confusing. When you check a task, you render the tasks once again
+  const checkTaskHandler = (task, isChecked) => {
+    task.checked = isChecked;
+    renderTasks(project, checkTaskHandler);
+  };
+  renderTasks(project, checkTaskHandler);
+};
+
```

Comment:
This one is a tricky one. That's what I came up with to avoid a bug.

So what happens, the methods calls for renderTasks. renderTasks requires a `callback` for `checkTaskHandler`. This callback will have as parameter task (the task in question) and isChecked (true or false).
The role of the callback will be to update the state and trigger a new render so the html changes with the new state state.

But when a task is checked the state changes and requires to retrigger a render. The render that requires the `checkTaskHandler`. So you can see the usage of `checkTaskHandler` at line 23 and 25.